iroh-sync/src/store.rs

```diff
@@ -180,9 +180,10 @@ pub trait Store: std::fmt::Debug + Clone + Send + Sync + 'static {
 #[derive(Debug, Default)]
 pub struct QueryBuilder<K> {
     kind: K,
-    filter_author: AuthorMatcher,
-    filter_key: KeyMatcher,
-    limit_offset: LimitOffset,
+    filter_author: AuthorFilter,
```

Comment:
any specific reasoning for Matcher -> Filter?

Reply:
No special reason. Can revert. I think I renamed some variables to `filter` and then thought it to match better with `Filter`. But no preference on my side I think.

Reply:
I think filter is fine, was just curious